### PR TITLE
Add part level.

### PR DESCRIPTION
**Pull request by martinslabber**
The idea is to have a level method on the part, that acts as a shortcut for several calls to level and subtract on the assembly.
Sections like this:
```
    box.level(ccbl.top, top_side.bottom)
    box.level(ccbl.back, top_side.back)
    box.level(ccbl.right, right_side.left)
    box.subtract(top_side.bottom, ccbl)
    box.subtract(right_side.left, ccbl)
```
Becomes:
```
ccbl.level(back=top_side.back)
ccbl.level(top=top_side.bottom, right=right_side.left, subtract=True)
```
If all 3 the sides had a subtract then it could have been one line.

### CODE REVIEW

**Comment by martinslabber** [comment]
At the time of creation the PR is not tested or well thought through but I'd like @HelenForbes to have a look and see if the overall idea makes sense.
I have a similar idea for rotate for another PR.

**Comment by HelenForbes** [comment]
I made a mistake and pushed to this branch instead of adding a new one. Once you've done the git add * it doesn't seem like you can ratther commit is to a different branch.

File: tests/shared.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ def hex_code_check(tmp_path: Path, filename: str, ext: str, hex_code: str = "",
     if return_hex:
         return hash_value_file
     else:
+        print(f"Got {hash_value_file} expect {hex_code}")
```
Comment (martinslabber) [inline tests/shared.py:16]: The test will only show this message when it fails.

**Review by HelenForbes — APPROVED** [review]

**Comment by martinslabber** [comment]
I ran the full test suite on this branch and all checked out good.

**Pull request merged**